bitcoin/script/miniscript.h

```diff
@@ -1014,224 +1014,384 @@ bool DecomposeScript(const CScript& script, std::vector<std::pair<opcodetype, st
+                if (k < 1 || k > n) return {};
+                in += 3 + n;
+                std::reverse(keys.begin(), keys.end());
+                constructed.push_back(MakeNodeRef<Key>(NodeType::MULTI, std::move(keys), k));
+            }
+            // c: wrapper
+            else if (in[0].first == OP_CHECKSIG) {
+                ++in;
+                to_parse.emplace_back(DecodeContext::Check, -1, -1);
+                to_parse.emplace_back(DecodeContext::Expression, -1, -1);
+            }
+            // v: wrapper
+            else if (in[0].first == OP_VERIFY) {
+                ++in;
+                to_parse.emplace_back(DecodeContext::Verify, -1, -1);
+                to_parse.emplace_back(DecodeContext::Expression, -1, -1);
+            }
+            // n: wrapper
+            else if (in[0].first == OP_0NOTEQUAL) {
+                ++in;
+                to_parse.emplace_back(DecodeContext::ZeroNotEqual, -1, -1);
+                to_parse.emplace_back(DecodeContext::Expression, -1, -1);
+            }
+            // Thresh
+            else if (last - in >= 3 && in[0].first == OP_EQUAL && ParseScriptNumber(in[1], k)) {
+                if (k < 1) return {};
+                in += 2;
+                to_parse.emplace_back(DecodeContext::ThreshW, 0, k);
+            }
+            // a: wrapper
+            else if (in[0].first == OP_FROMALTSTACK) {
+                ++in;
+                to_parse.emplace_back(DecodeContext::Alt, -1, -1);
+                to_parse.emplace_back(DecodeContext::MaybeAndV, -1, -1);
+                to_parse.emplace_back(DecodeContext::MaybeSwap, -1, -1);
+                to_parse.emplace_back(DecodeContext::Expression, -1, -1);
+            }
+            // OP_ENDIF can be WRAP_J, WRAP_D, ANDOR, OR_C, OR_D, or OR_I
+            else if (in[0].first == OP_ENDIF) {
+                ++in;
+                to_parse.emplace_back(DecodeContext::EndIf, -1, -1);
+                to_parse.emplace_back(DecodeContext::MaybeAndV, -1, -1);
+                to_parse.emplace_back(DecodeContext::MaybeSwap, -1, -1);
+                to_parse.emplace_back(DecodeContext::Expression, -1, -1);
+            }
+            // and_b
+            else if (in[0].first == OP_BOOLAND) {
+                ++in;
+                to_parse.emplace_back(DecodeContext::AndB, -1, -1);
+                to_parse.emplace_back(DecodeContext::Expression, -1, -1);
+                to_parse.emplace_back(DecodeContext::MaybeSwap, -1, -1);
+                to_parse.emplace_back(DecodeContext::Expression, -1, -1);
+            }
+            // or_b
+            else if (in[0].first == OP_BOOLOR) {
+                ++in;
+                to_parse.emplace_back(DecodeContext::OrB, -1, -1);
+                to_parse.emplace_back(DecodeContext::Expression, -1, -1);
+                to_parse.emplace_back(DecodeContext::MaybeSwap, -1, -1);
+                to_parse.emplace_back(DecodeContext::Expression, -1, -1);
```

Comment:
Because of the above observation, note that we don't do a push for `DecodeContext::MaybeAndV` for the first argument of and_b, or_b. But we do a `DecodeContext::MaybeAndV` for the second arg of and_b, or_b and fragments in OP_ENDIF (because they are guaranteed to not be the first argument).